compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -0,0 +1,157 @@
+	BackoffFilePath     string
+	BackoffWaitCallback func(time.Duration)
+}
+
+// TriggerNodeInventory creates a Node Inventory to send to Sensor.
+// To reduce strain on the Node, a linear backoff is checked before collecting an inventory.
+// Additionally, a cached inventory from an earlier invocation may be used instead of a full inventory run if it is fresh enough.
+// Note: This does not prevent strain in case of repeated pod recreation, as both mechanisms are based on an EmptyDir.
+func TriggerNodeInventory(opts *InventoryScanOpts) (*sensor.MsgFromCompliance, error) {
+	// check whether a cached inventory exists that is recent enough to use
+	cachedInventory := loadCachedInventory(opts)
+	if cachedInventory != nil && isCachedInventoryValid(cachedInventory) {
+		log.Debugf("Using cached scan created at %v", cachedInventory.GetScanTime())
+		return createMessage(cachedInventory), nil
+	}
+
+	// check for existing backoff, wait for specified duration if needed, then persist the new backoff duration
+	initialBackoff := env.NodeScanInitialBackoff.DurationSetting()
+	currentBackoff, err := getCurrentBackoff(opts.BackoffFilePath)
+	if err != nil {
+		// Set to maxBackoff to make sure scanning still happens, even if the backoff file gets corrupted
+		*currentBackoff = env.NodeScanMaxBackoff.DurationSetting()
+	}
+	if *currentBackoff > initialBackoff {
+		log.Warnf("Found existing backoff - last scan may have failed. Waiting %v seconds before retrying", currentBackoff.Seconds())
+		opts.BackoffWaitCallback(*currentBackoff)
+	}
+	writeBackoff(calcNextBackoff(*currentBackoff), opts.BackoffFilePath)
+
+	// if no inventory exists, or it is too old, collect a fresh one and save it to the cache
+	newInventory, err := opts.Scanner.Scan(opts.NodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	err = persistInventoryToCache(newInventory, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "persisting inventory to cache")
+	}
+
+	// Remove backoff directly before returning message, so that a failing/killed container does not lead to
+	// frequent rescans of a Node, which are costly and might impact Node performance
+	removeBackoff(opts.BackoffFilePath)
+	return createMessage(newInventory), nil
+}
+
+func getCurrentBackoff(backoffFilePath string) (*time.Duration, error) {
+	backoffInterval := env.NodeScanInitialBackoff.DurationSetting()
+
+	backoffFileContents, err := os.ReadFile(backoffFilePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			log.Debug("No backoff found, continuing without pause")
+		} else {
+			// return
+			return nil, err
+		}
+	} else {
+		// We have an existing backoff counter
+		backoffInterval, err = time.ParseDuration(string(backoffFileContents))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &backoffInterval, nil
+}
+
+func writeBackoff(backoff time.Duration, path string) {
+	err := os.WriteFile(path, []byte(backoff.String()), 0600)
+	if err != nil {
+		log.Warnf("Error writing backoff marker: %v", err)
```

Comment:
"marker" was yesterday. Today it's "file".
Seriously, I wasn't able to give it a good name in the last round, but simply "file" may be less confusing than "marker".

```suggestion
		log.Warnf("Error writing node scan backoff file: %v", err)
```